code/player/TTTPlayer.Lookup.cs

```diff
@@ -8,20 +8,32 @@ partial class TTTPlayer
 
         private const float MAX_DRAW_DISTANCE = 500;
 
-        private TTTPlayer IsLookingAtPlayer()
+        public T IsLookingAtType<T>(float distance)
         {
-            TraceResult trace = Trace.Ray(EyePos, EyePos + EyeRot.Forward * MAX_DRAW_DISTANCE)
-                .Ignore(ActiveChild)
-                .Ignore(this)
-                .UseHitboxes()
-                .Run();
+            Sandbox.Camera camera = Camera as Sandbox.Camera;
+
+            Trace trace = Trace.Ray(camera.Pos, camera.Pos + camera.Rot.Forward * distance)
+                .HitLayer(CollisionLayer.Debris)
+                .Ignore(this);
+
+            if (IsSpectatingPlayer)
+            {
+                trace.Ignore(CurrentPlayer);
+            }
 
-            if (trace.Hit && trace.Entity is TTTPlayer target)
+            TraceResult tr = trace.Run();
+
+            if (tr.Hit && tr.Entity is T type)
             {
-                return target;
+                return type;
             }
 
-            return null;
+            return default(T);
+        }
+
+        private TTTPlayer IsLookingAtPlayer()
+        {
+            return IsLookingAtType<TTTPlayer>(MAX_DRAW_DISTANCE);
```

Comment:
Do we want to make a IsLookingAtPlayerCorpse? Don't really mind if not. The interface cleans it up a bunch in general.

Reply:
Wouldn't do so as the corresponding distance is saved in another file :o